### DISCUSSION TITLE
Variogram calculation for point set

### DISCUSSION THREAD

**Discussion by wtrainor** [GeoStat-Framework/GSTools#175] [Q&A] [answered]
Hello,

I'm confused about how you would best calculate the experimental variogram of a point set data. The examples I found for your library always seems to generate an exhaustive field first (gs.SRF) which uses an input variogram.  If I have (x, y, property), can  I just calculate on those? I assume I would do it with a masked array, so I still need to define a grid. Is that correct?

Thank you!
Whitney

**Comment by wtrainor** [comment]
Nevermind, maybe?  I just have to use
`mesh_type="unstructured"`
in `gs.vario_estimate`
right?

**Comment by LSchueler** [comment, marked as answer]
Hello again!
Exactly, in [this example](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/03_variogram/00_fit_variogram.html#sphx-glr-examples-03-variogram-00-fit-variogram-py) we generate a point cloud of values, without a grid. The `mesh_type="unstructured"` is the default value, that's why you don't see that argument in the example.

**Reply by MuellerSeb** [reply]
Also the bins can be calculated automatically. So if you got `(x, y, property)`, you can simply call:
```python
import gstools as gs
bin_center, gamma = gs.vario_estimate((x, y), property)
```
And you get what you want :blush:
